internal/datastore/common/gc.go

```diff
@@ -73,6 +73,10 @@ func RegisterGCMetrics() error {
 // GarbageCollector represents any datastore that supports external garbage
 // collection.
 type GarbageCollector interface {
+	HasGCRun() bool
+	MarkGCCompleted()
+	ResetGCCompleted()
```

Comment:
given this is only used in tests, can we perhaps extract it into its own interface. Then we can do type assertions in the test based on that interface. And the interface does not even need to be here, but closer to the consumer in the  test package using it

Reply:
I prefer it being here to ensure that *all* GCable datastores are *forced* to implement it

Reply:
it can be forced if you make the test fail if the datastore does not implement the interface. Why making the API surface larger for something only used in a single test?

Reply:
Because not all datastores implement the custom GC code